etc/systemd/system/zfs-import-cache.service.in

```diff
@@ -5,10 +5,11 @@ Requires=systemd-udev-settle.service
 After=systemd-udev-settle.service
 After=cryptsetup.target
 Before=dracut-mount.service
-ConditionPathExists=@sysconfdir@/zfs/zpool.cache
+ConditionPathExists=${ZPOOL_CACHE}
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
+EnvironmentFile=-@initconfdir@/zfs
 ExecStartPre=/sbin/modprobe zfs
-ExecStart=@sbindir@/zpool import -c @sysconfdir@/zfs/zpool.cache -aN
+ExecStart=@sbindir@/zpool import ${ZPOOL_CACHE:+-c "${ZPOOL_CACHE}"} -aN ${ZPOOL_IMPORT_OPTS}
```

Comment:
We'll need to do something similar here or we'll break all the systems which don't have this file installed.